zim/gui/pageview/__init__.py

```diff
@@ -53,7 +53,7 @@
 	Dialog, FileDialog, QuestionDialog, ErrorDialog, \
 	IconButton, MenuButton, BrowserTreeView, InputEntry, \
 	ScrolledWindow, \
-	rotate_pixbuf, populate_popup_add_separator, strip_boolean_result, \
```

Comment:
When I started working, I didn't read this line. 🤦‍♂️

https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/1ccaffc181671a71b6d66cbe99ae14688db1fc6c/README.md?plain=1#L41

``GdkPixbuf.Pixbuf.apply_embedded_orientation`` available since **2.12** 🤦‍♂️

therefore, this code:

https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/1ccaffc181671a71b6d66cbe99ae14688db1fc6c/zim/gui/widgets.py#L341-L361

is completely equivalent to this сode
```python3
def rotate_pixbuf(pixbuf):
	'''Rotate the pixbuf to match orientation from EXIF info.
	This is intended for e.g. photos that have EXIF information that
	shows how the camera was held.
	@returns: a new version of the pixbuf or the pixbuf itself.
	'''
	return GdkPixbuf.Pixbuf.apply_embedded_orientation(pixbuf)
```

Currently, ``rotate_pixbuf`` is used in two places.

1. https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/e4b04df4737cde00d0c159b1007361204fa7856c/zim/gui/pageview/__init__.py#L514
2. https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/1ccaffc181671a71b6d66cbe99ae14688db1fc6c/zim/plugins/attachmentbrowser/thumbnailer.py#L102

We are currently working on the first one.

And if we replacing the second place with  
```python3
pixbuf = GdkPixbuf.Pixbuf.apply_embedded_orientation(pixbuf)
```
we can completely remove ``rotate_pixbuf``.

Can we do that?

Reply:
I believe you're correct, nice catch! We can indeed retire our `rotate_pixbuf` implementation. `return GdkPixbuf.Pixbuf.apply_embedded_orientation(pixbuf)` plus a `logger.warning('Using deprecated function "zim.XXX.rotate_pixbuf" - please update your code to use "GdkPixbuf.Pixbuf.apply_embedded_orientation" instead')`